ONSdigital/eq-author-app
# Ear-226 date range duration validation

> BEFORE MAKING YOUR PR... There must be no linting errors and all tests must pass

### What is the context of this PR?

Added validation to duration on date range questions.

This was a bit of a pain in the bum because all the validations were generated dynamically and didn't include a way to include another property error. 

Had to separate the way the date range validation buttons are generated. It felt a bit messy when I did it so would appreciate any comments on the front end part.

The validator logic makes no sense to me but for some reason it works.

### How to review 

> Add to the list below as appropriate, including screenshots when necessary

1. Create a new questionnaire with a date range question
2. Add validation for both earliest/latest date and duration
3. Make the max bigger than the min
4. Error messages should appear for both

Look out for:
- Validation rendering in order as expected
- Error messages disappearing when conditions are corrected
- Error messages actually appearing
- Date questions working
- Multiple various questions working
